src/ripple/nodestore/DatabaseShard.h

```diff
@@ -211,6 +211,14 @@ class DatabaseShard : public Database
     static constexpr std::uint32_t ledgersPerShardDefault {16384u};
 };
 
+constexpr
+std::uint32_t
+seqToShardIndex(std::uint32_t seq,
+    std::uint32_t ledgersPerShard = DatabaseShard::ledgersPerShardDefault)
+{
+    return (seq - 1) / ledgersPerShard;
+}
+
```

Comment:
`PeerImp::hasLedger` in PeerImp.cpp:380 might be an additional place to use this new function.

Reply:
Indeed but I didn't want to muddy this PR with it. I will be pushing a separate PR of related miscellaneous fixes that include your suggestion. Thanks!